electrum/gui/qt/main_window.py

```diff
@@ -415,12 +459,42 @@ def on_error(self, exc_info):
     def on_event_wallet_updated(self, wallet):
         if wallet == self.wallet:
             self.need_update.set()
+            if self.password_future:
+                self.password_future.cancel()
+                self.password_future = None
 
     @event_listener
     def on_event_new_transaction(self, wallet, tx):
         if wallet == self.wallet:
             self.tx_notification_queue.put(tx)
 
+    @qt_event_listener
+    def on_event_password_required(self, wallet, future, message):
+        if wallet == self.wallet:
+            self.password_future = future
+            self.show_action_required(future, message)
+
+    @qt_event_listener
+    def on_event_password_not_required(self, wallet):
+        if wallet == self.wallet:
+            self.password_future = None
```

Comment:
Shouldn't `password_future` be cancelled here (conditionally if `not None`) before clearing?